pkg/ocihook/state/state.go

```diff
@@ -0,0 +1,89 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package state
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/containerd/nerdctl/v2/pkg/lockutil"
+)
+
+// This is meant to store stateful informations about containers that we receive from ocihooks
+// We are storing them inside the container statedir
+// Note that you MUST use WithLock to perform any operation (like Read or Write).
+// Typically:
+// lf.WithLock(func ()error {
+//   lf.Load()
+//   // Modify something on the object
+//   lf.StartedAt = ...
+//   lf.Save()
+// })
+
+const (
+	lifecycleFile = "lifecycle.json"
+)
+
+func NewLifecycleState(stateDir string) *LifecycleState {
+	return &LifecycleState{
+		stateDir: stateDir,
+	}
+}
+
+type LifecycleState struct {
+	stateDir  string
+	StartedAt int64 `json:"started_at"`
```

Comment:
```suggestion
	StartedAt time.Time `json:"started_at"`
```

Reply:
Ok.
Last version now serializes and unserialize `time.Time` instead of timestamp.

@AkihiroSuda PTAL